samples/boards/nrf52/mesh/onoff_level_lighting_vnd_app/CMakeLists.txt

```diff
@@ -19,4 +20,5 @@ zephyr_include_directories(
 	src/
 	src/mesh
 	$ENV{ZEPHYR_BASE}/subsys/bluetooth/controller/hal
+	$ENV{ZEPHYR_BASE}/subsys/bluetooth/host
```

Comment:
@vikrant8051 your application exists to serve as reference code for developers. I don't think it's a good idea to encourage people to include internal headers and use internal APIs. If you want to hack around feel free to do it for a private project, but I don't think we should be putting such things in sample apps that are part of the upstream tree.

The line above related to controller/hal - what's that for?

Reply:
Sorry, I was not aware about it. But there is no proper reference (for how to settings layer API in App), so I've to do it assuming one of the reviewer will correct me :)

Ok... if I removed $ENV{ZEPHYR_BASE}/subsys/bluetooth/host from CMakeLists.txt & #include "settings.h" from storage.c I'm getting following error:



../src/storage.c:16:11: warning: implicit declaration of function 'BT_SETTINGS_SIZE'; did you mean 'BT_BUF_RX_SIZE'? [-Wimplicit-function-declaration]
  char buf[BT_SETTINGS_SIZE(sizeof(reset_counter))];
           ^~~~~~~~~~~~~~~~
           BT_BUF_RX_SIZE
In file included from /home/vikrant/zephyr/include/misc/slist.h:20:0,
                 from /home/vikrant/zephyr/include/settings/settings.h:12,
                 from ../src/mesh/ble_mesh.h:11,
                 from ../src/storage.c:8:
../src/storage.c: At top level:
../src/storage.c:58:32: error: expected ')' before '(' token
 BT_SETTINGS_DEFINE(ps, ps_set, NULL, NULL);
                                ^
../src/storage.c:42:12: warning: 'ps_set' defined but not used [-Wunused-function]
 static int ps_set(int argc, char **argv, char *val)
            ^~~~~~
Kindly guide me to improve it. Thank You !!

Reply:
What API did you think I'm talking about when I said "internal API"? I'm talking about BT_SETTINGS. If something is not exposed through a header file in include/bluetooth it's an internal API which is not designed to be used by applications directly.

What's preventing you from choosing some other sub-path than "bt/" and using the settings API directly instead of BT_SETTINGS (which is just a wrapper on top of settings for managing stuff under "bt/")?

Reply:
@vikrant8051 you could look at subsys/bluetooth/host/settings.c and see how it uses the settings API, and then do something similar (probably simpler) for your app where you use some other sub-path. The reason BT_SETTINGS is not public is that the Bluetooth stack "reserves" for itself everything under the "bt/" path, i.e. it needs to be able to trust that there's nothing else there that can cause conflicts, both now and in future versions of the stack.

Reply:
Ok

Reply:
@Laczen 
Hey under documentation there is now details about what I'm looking for
https://docs.zephyrproject.org/latest/subsystems/settings/settings.html

Your mailing list answer, actually opens new window for me to go one step further.
So it was great help from your side, since first time I could successfully implement 
what I want using settings_layer. :) :+1:

Reply:
> I think the settings_load needs to be called only in the main to allow other subsystems to use the settings subsystem.

@Laczen the only requirement from the Bluetooth side is that settings_load() is called after the stack has completed initialization. You can do it from main() but then you should either call bt_enable() synchronously (NULL callback) or have some other way to block in main() while waiting for bt_ready() to be called.